src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -216,14 +217,30 @@ protected String buildScriptToExecuteString(InstanceScript instanceScript) {
     }
 
     private RunScriptOptions buildScriptOptions(InstanceScript instanceScript) {
-        logger.info("Script options: userLogin=" + getVmUserLogin());
-        return Optional.ofNullable(instanceScript.getCredentials())
-                       .map(credentials -> RunScriptOptions.Builder.runAsRoot(false)
-                                                                   .overrideLoginCredentials(new LoginCredentials.Builder().user(credentials.getUsername())
-                                                                                                                           .password(credentials.getPassword())
-                                                                                                                           .authenticateSudo(false)
-                                                                                                                           .build()))
-                       .orElse(RunScriptOptions.Builder.overrideLoginUser(getVmUserLogin()));
+        return Optional.ofNullable(instanceScript.getCredentials()).map(credentials -> {
+            if (credentials.getPrivateKey() != null) {
+                String username = Optional.of(credentials.getUsername())
+                                          .filter(StringUtils::isNotEmpty)
+                                          .filter(StringUtils::isNotBlank)
```

Comment:
Nice tips! Correct me if I'm wrong but it seems this will trigger a NPE if `credentials.getUsername()` is null. Using `Optional.ofNullable` instead will be NPE safe.

Reply:
nice spot thanks